Server/mods/deathmatch/logic/CAccount.cpp

```diff
@@ -46,8 +46,9 @@ void CAccount::SetName(const std::string& strName)
 {
     if (m_strName != strName)
     {
-        m_pManager->ChangingName(this, m_strName, strName);
-        m_strName = strName;
+        std::string strNewName = strName.substr(0, MAX_USERNAME_LENGTH);
+        m_pManager->ChangingName(this, m_strName, strNewName);
+        m_strName = strNewName;
```

Comment:
You don't use the value anymore, we can move it and avoid a copy (if the compiler doesn't avoid it).
```suggestion
        m_strName = std::move(strNewName);
```